File: pattern-02-classical-fallback/skeleton/java/com/csxplore/quantumpatterns/fallback/Application.java
package com.csxplore.quantumpatterns.fallback;

import com.csxplore.quantumpatterns.fallback.adapter.ClassicalSolver;
import com.csxplore.quantumpatterns.fallback.adapter.FallbackSolver;
import com.csxplore.quantumpatterns.fallback.adapter.QuantumSolver;
import com.csxplore.quantumpatterns.fallback.port.Solver;
import com.csxplore.quantumpatterns.fallback.service.OptimizationService;

/**
 * Demonstrates automatic fallback execution.
 */
public class Application {

    public static void main(String[] args) {

        Solver solver =
                new FallbackSolver(
                        new QuantumSolver(),
                        new ClassicalSolver());

        OptimizationService service =
                new OptimizationService(solver);

        System.out.println(
                service.optimize("Route-101").solverName());

    }

}

File: pattern-02-classical-fallback/skeleton/python/application.py
"""
Educational demonstration.
"""

from request import OptimizationRequest
from quantum_solver import QuantumSolver
from classical_solver import ClassicalSolver
from fallback_solver import FallbackSolver


solver = FallbackSolver(
    QuantumSolver(),
    ClassicalSolver()
)

result = solver.solve(
    OptimizationRequest("Route-101")
)

print(result.solver_name)